### PR TITLE
Reset token type of meta.embedded scopes

### PR DESCRIPTION
**Pull request by mjbvz**
Fixes https://github.com/Microsoft/vscode/issues/34857

See https://github.com/Microsoft/vscode/issues/34857 for more details about this issue

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -54,7 +54,8 @@ export const enum StandardTokenType {
 	Other = 0,
 	Comment = 1,
 	String = 2,
-	RegEx = 4
+	RegEx = 4,
+	MetaEmbedded = 8
```
Comment (mjbvz) [inline src/main.ts]: No encoded tokens will ever be marked as MetaEmbeded. It's just used internally to reset the scope. Let me know if you can think of a better way to express this

**Comment by mjbvz** [comment]
Hey @alexandrudima, can you please take a look at this PR and let me know if you have any feedback or alternate design suggestions

**Comment by mjbvz** [comment]
Ping @alexandrudima. I'd really like to see if we can address https://github.com/Microsoft/vscode/issues/34857 for October

File: src/main.ts (hunk 1 above)
Comment (alexandrudima) [reply src/main.ts]: Yes, this is problematic because this enum makes it in the public module API

**Comment by mjbvz** [comment]
The changes to use the internal enum type look good to me. Thanks for looking into this

**Pull request merged**